### PR TITLE
Added: option to set a level for showing notifications

### PR DESCRIPTION
**Pull request by MarcelRoozekrans**
### Functional description

Added an option to set the minimum level for showing notfications to the end-user. 

### Reasoning

With this change the end-user can deside on what notifications are showing for retrospect.
Defaulting to info showing all messages.

### Technical description

- Added a setting, in normal setting and template
- Implemented setting in AddonSettings
- Hiding actual messages in XbmcWrapper

### CODE REVIEW

File: resources/lib/addonsettings.py [hunk 1]
```diff
@@ -650,6 +650,23 @@ def get_release_track():
...
+    @staticmethod
+    def get_notification_level():
+        """ Retrieves the level for displaying notifications to the end user:
+
+        0: Information.
+        1: Warning.
+        2: Error.
+
+        :return: Array with the levels that we are showing to the end user
+        :rtype: array
```
Comment (basrieter) [inline resources/lib/addonsettings.py:662]: Could you make this `:rtype: list[str]` that way it helps PyCharm a bit better.

Comment (MarcelRoozekrans) [reply resources/lib/addonsettings.py:662]: Sure, I use PyCharm aswell can we also add the .idea folder to the .gitignore ?

**Comment by MarcelRoozekrans** [comment]
> First of: Thanks for the contribution! Never expected people to find them annoying. The ones that are "Info" level are:
> 
> * "Starting Retrospect"
> * "Updating channels"
> * "An update is available"
> 
> The first one only happens once per session. the second one happens after installation of a new Retrospect and the latter is only present if you did not install from the Kodi Repository.
> 
> But hey, the change looks nice. I have one minor thing. Hope you can fix that

Yeah, The annoyance is when I use this as a widget every time my shield wakes up I get the retrospect startup notification.

**Comment by basrieter** [comment]
Thanks for your contribution.

**Pull request merged**